versions/3.0.md

```diff
@@ -196,9 +205,73 @@ An object representing a Host.
 
 Field Name | Type | Description
 ---|:---:|---
-<a name="oasHost"></a>host | `string` | The host (name or ip) serving the API. This MUST be the host only and does not include the scheme nor sub-paths. It MAY include a port. If the `host` is not included, the host serving the documentation is to be used (including the port). The `host` does not support [path templating](#pathTemplating).
-<a name="oasBasePath"></a>basePath | `string` | The base path on which the API is served, which is relative to the [`host`](#oasHost). If it is not included, the API is served directly under the `host`. The value MUST start with a leading slash (`/`). The `basePath` does not support [path templating](#pathTemplating). 
-<a name="oasScheme"></a>scheme | `string` | The transfer protocol of the API. Values MUST be from the list: `"http"`, `"https"`, `"ws"`, `"wss"`. If the `scheme` is not included, the default scheme to be used is the one used to access the OpenAPI definition itself.
+<a name="hostUrlTemplate"></a>host URL template | `string` | A URL to the target host.  This URL supports template variables and may be relative, to indicate that the host location is relative to the location where the OpenAPI Specification is being served.  Templates are _optional_ and specified by the #hostTemplateParameter syntax.  Template substitutions will be made when a variable is named in `{`brackets`}`.
+<a name="hostDescription"></a>host description | `string` | An optional string describing the host designated by the URL.
+<a name="hostTemplates"></a>templates | [Templates Object](#hostTemplatesObject) | An object holding templates for substitution in the URL template
+
+This object can be extended with [Specification Extensions](#specificationExtensions). 
+
+##### Host Object Example
+
+The following shows how multiple hosts can be described in the host object array
+
+```yaml
+servers:
+- url: https://development.gigantic-server.com/v1
+  description: Development server
+- url: https://staging.gigantic-server.com/v1
+  description: Staging server
+- url: https://api.gigantic-server.com/v1
+  description: Production server
+```
+
+The following shows how templates an be used for a host configuration
+
+```yaml
+servers:
+- url: {scheme}://{host}:{port}/{basePath}
+  description: The production API server
+  templates:
+    scheme:
+      enum:
+        - https
+        - http
+      default: https
```

Comment:
`default` currently applies to optional parameters and represents the server-assumed default, like the number of results in a page or an offset. By applying it to a required element, we confuse an already tricky aspect of a spec. An alternative could be `example`.